cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -404,44 +402,61 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
             return parseInt(this.$('#due_in').val());
         },
 
+        showProjectedDate: function() {
+            var startDate = DateUtils.parseDateFromString(this.model.get('start'));
+            startDate = new Date(startDate.getUTCFullYear(), startDate.getUTCMonth(), startDate.getUTCDate());
+            this.$("#relative_weeks_due_start_date").text(startDate.toDateString());
+            var projectedDate = new Date(startDate)
+            projectedDate.setDate(projectedDate.getDate() + this.getValue()*7);
+            projectedDate = new Date(projectedDate.getUTCFullYear(), projectedDate.getUTCMonth(), projectedDate.getUTCDate())
+            this.$("#relative_weeks_due_projected_due_in").text(projectedDate.toDateString());
+            if (startDate <= projectedDate) {
+                this.$('#relative_weeks_due_projected').show();
+            }
+        },
+
         validateDueIn: function() {
             if (this.getValue() > 18){
-                this.$('#due-num-weeks-warning-max').show();
+                this.$('#relative_weeks_due_warning_max').show();
+                this.$('#relative_weeks_due_projected').hide();
                 BaseModal.prototype.disableActionButton.call(this.parent, 'save');
             }
             else if (this.getValue() < 1){
-                this.$('#due-num-weeks-warning-min').show()
+                this.$('#relative_weeks_due_warning_min').show()
+                this.$('#relative_weeks_due_projected').hide();
                 BaseModal.prototype.disableActionButton.call(this.parent, 'save');
             }
             else {
-                this.$('#due-num-weeks-warning-max').hide();
-                this.$('#due-num-weeks-warning-min').hide();
+                this.$('#relative_weeks_due_warning_max').hide();
+                this.$('#relative_weeks_due_warning_min').hide();
+                this.$('#relative_weeks_due_projected').hide();
+                if (this.model.get('start')){
+                    this.showProjectedDate();
+                }
                 BaseModal.prototype.enableActionButton.call(this.parent, 'save');
             }
         },
 
-        clearValue: function(event) {
-            event.preventDefault();
-            this.$('#due_in').val('');
-        },
-
         afterRender: function() {
             AbstractEditor.prototype.afterRender.call(this);
-            this.$('.field-due-in input').val(this.model.get('due_num_weeks'));
+            this.$('.field-due-in input').val(this.model.get('relative_weeks_due'));
+            this.$('#relative_weeks_due_projected').hide();
```

Comment:
I need to explicitly hide it in my current implementation because otherwise I would just get `If a learner starts on , this subsection will be due on .` (this text is not a warning message class so the default isn't hidden).

Reply:
Ah makes sense. Thank you!